main.go

```diff
@@ -138,6 +143,33 @@ func (c *CouchStore) GetStoreObj(jsonObj []byte) (error, caddyshack.StoreObject)
 	return nil, obj
 }
 
+func (c *CouchStore) MarshalStoreObjects(data []byte) (result []caddyshack.StoreObject, err error) {
```

Comment:
Hey Actually its a slice of pointers thats returned. The caddyshack.StoreObject is an interface which implements a pointer. The getStoreObj actually creates the type from the type that's used to register with the store. So in essence its a slice of pointers.